axlearn/cloud/gcp/pathways_utils.py

```diff
@@ -414,9 +414,17 @@ def _build_pathways_head_pod(self) -> Nested[Any]:
                 }
             )
 
+        # node_selector = {
+        #     _PATHWAYS_HEAD_NODE_POOL_SELECTOR_KEY: _PATHWAYS_HEAD_NODE_POOL_SELECTOR_VALUE,
+        # }
+
+        system = USER_FACING_NAME_TO_SYSTEM_CHARACTERISTICS[self._tpu_type]
+        #pod_spec["nodeSelector"].update(
         node_selector = {
-            _PATHWAYS_HEAD_NODE_POOL_SELECTOR_KEY: _PATHWAYS_HEAD_NODE_POOL_SELECTOR_VALUE,
-        }
+                "cloud.google.com/gke-tpu-accelerator": system.gke_accelerator,
```

Comment:
I think this won't work as expected when you have many nodepools that match this. We need to ensure the patways head lands on the same TPU VM as the other parts of the jobset.